File: elms-frontend/src/services/EmployeeService.js
import API from "./ApiService";

export const getAllEmployees = () => {
    return API.get("/employee");
}

export const deleteEmployee = (id) => {
    return API.delete(`/employee/${id}`);
};

export const getEmployeeById = (id) => {
    return API.get(`/employee/${id}`);
};

export const updateEmployee = (employee) => {
    return API.put("/employee", employee);
};

export const saveEmployee = (employee) => {
    return API.post("/employee", employee);
};

File: backend/src/main/java/com/jsp/elms/service/EmployeeService.java
package com.jsp.elms.service;

import java.util.List;

import com.jsp.elms.entity.Employee;

public interface EmployeeService {

    public Employee saveEmployee(Employee employee);

    public List<Employee> getAllEmployees();

    public Employee getEmployeeById(Integer id);

    public Employee updateEmployee(Employee employee);

    public void deleteEmployee(Integer id);
}
